.github/workflows/docker.yml

```diff
@@ -0,0 +1,58 @@
+name: Docker Build Action
+on:
+  pull_request:
+    branches:
+      - master
+  release:
+    types:
+      - published
+  push:
+    branches:
+      - master
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
+
+jobs:
+
+  build:
+    name: compile, build and push container
+    runs-on: ubuntu-latest
+    steps:
+
+    - name: Set up Go 1.22
+      uses: actions/setup-go@v5
+      with:
+        go-version: "1.22"
+        cache: false
+      id: go
+
+    - name: Check out code into the Go module directory
+      uses: actions/checkout@v4
+
+    # TODO: enable once all linter issues are solved
+    # - name: Lint
+    #   uses: golangci/golangci-lint-action@v4
+    #   with:
+    #     args: -p bugs -p unused --timeout=3m
+
+    - name: Log in to the Container registry
+      uses: docker/login-action@v3
+      with:
+        registry: ${{ env.REGISTRY }}
+        username: ${{ github.actor }}
+        password: ${{ secrets.GITHUB_TOKEN }}
+
+    - name: Make tag
+      run: |
+        [ "${GITHUB_EVENT_NAME}" == 'pull_request' ] && echo "tag=${GITHUB_HEAD_REF##*/}" >> $GITHUB_ENV || true
+        [ "${GITHUB_EVENT_NAME}" == 'release' ] && echo "tag=${GITHUB_REF##*/}" >> $GITHUB_ENV || true
+        [ "${GITHUB_EVENT_NAME}" == 'push' ] && echo "tag=latest" >> $GITHUB_ENV || true
+
+    - name: Build and push image
+      uses: docker/build-push-action@v5
+      with:
+        context: .
+        push: true
+        tags: ghcr.io/lightbitslabs/los-csi:${{ env.tag }}
```

Comment:
It appears you're tagging new image with the name of the branch  e.g
 lightbitslabs/los-csi:update-dependencies

Reply:
This is intentional, it tags the image according the name of the branch in case of a PR, once its in master a :latest tag is pushed, and once a tag is created, e.g. v1.16.0 the image is pushed with this tag

Reply:
do you delete those images after PR checks is passed ?  If not then we will have image for each PR and images will be adding up at the destination?

Reply:
No i do not delete the image. We never delete images from registries.

Reply:
My understating is images will keep piling up into registry for PRs, but i'll create a dummy PR and verify this for further appropriate action.

Reply:
Yes, but as only the layers of a image are stored which differ from others the total amount of storage consumed is negligible

Reply:
I'm good as long there is no limit on the max number of images in ghcr.io, or there is an auto policy to delete x-days old images.